src/classes/CRLP_Batch_Base.cls

```diff
@@ -344,4 +353,195 @@ public abstract class CRLP_Batch_Base {
+
+                if (customizableRollupSettings.Rollups_IncrementalLastNDays_ValOverride__c != null) {
+                    userDefinedLastNDays = (Integer) customizableRollupSettings.Rollups_IncrementalLastNDays_ValOverride__c;
+                }
+            }
+            return userDefinedLastNDays;
+        }
+        set;
+    }
+
+    /**
+     * @description API name of the date field to use as in the inner join query
+     * when the org has the user-defined inner join for incremental rollups
+     */
+    private String userDefinedLastNDaysField {
+        get {
+            if (userDefinedLastNDaysField == null) {
+                Customizable_Rollup_Settings__c customizableRollupSettings =
+                        UTIL_CustomSettingsFacade.getCustomizableRollupSettings();
+
+                if (String.isNotBlank(customizableRollupSettings.Rollups_IncrementalLastNDays_FldOverride__c)) {
+                    userDefinedLastNDaysField = customizableRollupSettings.Rollups_IncrementalLastNDays_FldOverride__c;
+                }
+            }
+            return userDefinedLastNDaysField;
+        }
+        set;
+    }
+
+    /**
+     * @description Produces the where clause that should be used for the inner join when NOT in incremental mode
+     * @return The WHERE clause, as a string, that should be used to filter the detail records
+     */
+    protected String getStandardWhereClause() {
+        String standardWhereClause = null; //will return null if this method is called from an unsupported rollup type
+        if (jobType == CRLP_RollupProcessingOptions.RollupType.AccountHardCredit
+                || jobType == CRLP_RollupProcessingOptions.RollupType.ContactHardCredit) {
+            standardWhereClause = 'Id IN (' +
+                    'SELECT ' + keyField + ' ' +
+                    'FROM ' + detailObjectType + ')';
+        } else if (jobType == CRLP_RollupProcessingOptions.RollupType.ContactSoftCredit) {
+            standardWhereClause = 'Id IN (SELECT ContactId FROM OpportunityContactRole)';
+        } else if (jobType == CRLP_RollupProcessingOptions.RollupType.AccountContactSoftCredit ||
+                jobType == CRLP_RollupProcessingOptions.RollupType.AccountSoftCredit) {
+            // It's not possible to do an inner query on this due to the following Salesforce error:
+            // "The inner select field 'Contact.AccountId' cannot have more than one level of relationships"
+        } else if (jobType == CRLP_RollupProcessingOptions.RollupType.RecurringDonations) {
+            standardWhereClause = 'Id IN (SELECT npe03__Recurring_Donation__c FROM Opportunity)';
+        }
+        return standardWhereClause;
+    }
+
+    /**
+     * @description Produces the where clause that should be used for the inner join
+     * @return The WHERE clause, as a string, that should be used to filter the detail records
+     */
+    @TestVisible protected String getWhereClause() {
```

Comment:
Since this is the main called outside this class to get the inner join portion for the main query locator, I think the method name should be something like `getMainQueryInnerJoinFilter()`.

Reply:
Ok, moved and renamed!